daemon/delete.go

```diff
@@ -128,6 +129,15 @@ func (daemon *Daemon) cleanupContainer(container *container.Container, forceRemo
 		return errors.Wrapf(err, "unable to remove filesystem for %s", container.ID)
 	}
 
+	for _, l := range container.HostConfig.Links {
+		_, alias, err := opts.ParseLink(l)
+		if err != nil {
+			return err
+		}
+		fullName := path.Join(container.Name, alias)
+		daemon.releaseName(fullName)
```

Comment:
@cpuguy83 These names are combined and registered in https://github.com/moby/moby/blob/master/daemon/daemon.go#L450, I just followed this code. Is there any problem in this way? Could you please give me more information?